### PR TITLE
[#3087] Add DB constraint that primary and alternative phonenumber differ

### PR DESCRIPTION
**Pull request by pi-sigma**
Taiga: https://taiga.maykinmedia.nl/project/open-inwoner/issue/3087

### CODE REVIEW

File: src/open_inwoner/accounts/migrations/0081_user_check_alternative_phonenumber_differs_from_primary_phonenumber.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Generated by Django 4.2.18 on 2025-03-11 15:35
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```
Comment (swrichards) [inline src/open_inwoner/accounts/migrations/0081_user_check_alternative_phonenumber_differs_from_primary_phonenumber.py:15]: Can we add a migration to ensure that duplicate numbers are consolidated before applying the migration?

Comment (pi-sigma) [reply src/open_inwoner/accounts/migrations/0081_user_check_alternative_phonenumber_differs_from_primary_phonenumber.py:15]: What's the best option, setting the alternative to "" in case of duplication?

Comment (swrichards) [reply src/open_inwoner/accounts/migrations/0081_user_check_alternative_phonenumber_differs_from_primary_phonenumber.py:15]: Yes that makes sense, that would be the default otherwise if it weren't set.

File: src/open_inwoner/accounts/forms.py [hunk 2]
```diff
@@ -194,6 +194,10 @@ def clean_phonenumber_alternative(self):
                     "A primary phone number is required for setting an alternative phone number"
                 )
             )
+        if phonenumber_alt and phonenumber_alt == phonenumber:
+            raise ValidationError(
+                _("Primary adn secondary phone numbers cannot be the same")
```
Comment (swrichards) [inline src/open_inwoner/accounts/forms.py]:
```suggestion
                _("Primary and secondary phone numbers cannot be the same")
```

**Review by swrichards — CHANGES_REQUESTED** [review]

File: src/open_inwoner/accounts/tests/test_user.py [hunk 3]
```diff
@@ -112,9 +112,16 @@ def test_plan_contact_new_count_methods(self):
...
 
-    def test_constraint_phonenumber_alternative(self):
+    def test_phonenumber_alternative_requires_primary_phonenumber(self):
         user = UserFactory(phonenumber="")
 
         with self.assertRaises(IntegrityError):
             user.phonenumber_alternative = "0612345678"
             user.save()
+
+    def test_phonenumber_alternative_differs_from_non_empty_primary_phonenumber(self):
```
Comment (swrichards) [inline src/open_inwoner/accounts/tests/test_user.py:122]: @pi-sigma sorry one more late thought: can we also add a test that confirms both phonenumbers can be empty?

Comment (pi-sigma) [reply src/open_inwoner/accounts/tests/test_user.py:122]: Good point, I'll add it

Comment (pi-sigma) [reply src/open_inwoner/accounts/tests/test_user.py:122]: I just added `phonenumber_alternative=""` to the `UserFactory` in one test to make it more explicit. I don't think this requires an extra test, since `UserFactory` will raise an `IntegrityError` upon creation if the constraint is violated.

Comment (swrichards) [reply src/open_inwoner/accounts/tests/test_user.py:122]: True, _but_ then we are counting on future refactorers (including ourselves) to be aware of that context, I'd still somewhat prefer having it as a separate test, even if small, to express that this is a specific invariant we care about.

Comment (pi-sigma) [reply src/open_inwoner/accounts/tests/test_user.py:122]: Fair enough, better explicit than implicit. I'll change it.

**Review by swrichards — APPROVED** [review]
